OrbitClientGgp/main.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ClientGgp.h"
+#include "OrbitBase/ThreadPool.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+
+ABSL_FLAG(uint64_t, grpc_port, 44765, "Grpc service's port");
+ABSL_FLAG(int32_t, pid, 0, "pid to capture");
+ABSL_FLAG(uint32_t, capture_length, 10, "duration of capture in seconds");
+ABSL_FLAG(uint16_t, sampling_rate, 1000,
+          "Frequency of callstack sampling in samples per second");
+ABSL_FLAG(bool, frame_pointer_unwinding, false,
+          "Use frame pointers for unwinding");
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+  uint64_t grpc_port = absl::GetFlag(FLAGS_grpc_port);
+  uint32_t capture_length = absl::GetFlag(FLAGS_capture_length);
+
+  if (!absl::GetFlag(FLAGS_pid)) {
+    FATAL("pid to capture not provided; set using -pid");
+  }
+
+  ClientGgpOptions options;
+  options.grpc_server_address = absl::StrFormat("127.0.0.1:%d", grpc_port);
+  options.capture_pid = absl::GetFlag(FLAGS_pid);
+
+  ClientGgp client_ggp(std::move(options));
+  if (!client_ggp.InitClient()) {
+    return -1;
+  }
+
+  // The request is done in a separate thread to avoid blocking main()
+  // It is needed to provide a thread pool
+  std::unique_ptr<ThreadPool> thread_pool =
+      ThreadPool::Create(1, 1, absl::Seconds(1));
```

Comment:
I am not sure about this initialisation, I only need one thread , is this the right way to do it (We need to use ThreadPool because is the one used in CaptureClient)?

Reply:
I think this is just fine as is; as you wrote you need the thread pool since it is in the signature of StartCapture.